src/Controls/src/Core/Shell/ShellContent.cs

```diff
@@ -212,9 +212,27 @@ Page ContentCache
 			shellContent.Route = Routing.GenerateImplicitRoute(pageRoute);
 
 			shellContent.Content = page;
-			shellContent.SetBinding(TitleProperty, new Binding(nameof(Title), BindingMode.OneWay, source: page));
-			shellContent.SetBinding(IconProperty, new Binding(nameof(Icon), BindingMode.OneWay, source: page));
-			shellContent.SetBinding(FlyoutIconProperty, new Binding(nameof(FlyoutIcon), BindingMode.OneWay, source: page));
+			shellContent.SetBinding(
+				TitleProperty,
+				TypedBinding.ForSingleNestingLevel(
+					nameof(TemplatedPage.Title),
+					static (TemplatedPage page) => page.Title,
+					mode: BindingMode.OneWay,
+					source: page));
+			shellContent.SetBinding(
+				IconProperty,
+				TypedBinding.ForSingleNestingLevel(
+					nameof(TemplatedPage.IconImageSource),
+					static (TemplatedPage page) => page.IconImageSource,
```

Comment:
The original bindings specified `"Icon"` and `"FlyoutIcon"` paths with a `TemplatedPage` source. `TemplatedPage` doesn't have either property. It only has `IconImageSource`. Was the previous implementation intentional and am I missing the point?